cpp/src/Ice/ConnectionI.cpp

```diff
@@ -397,10 +397,7 @@ Ice::ConnectionI::start(const StartCallbackPtr& callback)
             //
             // Wait for the connection to be validated.
             //
-            while(_state <= StateNotValidated)
-            {
-                wait();
-            }
+            _conditionVariable.wait(lock, [this]{ return _state >= StateNotValidated; });
```

Comment:
Is the equality check correct?
```suggestion
            _conditionVariable.wait(lock, [this]{ return _state > StateNotValidated; });
```

Reply:
Good catch!